File: examples/in_memory/frontend/events.js
class Event {
    constructor(type, payload) {
        this.type = type;
        this.payload = payload;
    }
}

class BroadcastMessageEvent {
    constructor(message, from) {
        this.message = message;
        this.from = from;
    }
}

class SendMessageEvent {
    constructor(message, from, to) {
        this.message = message;
        this.from = from;
        this.to = to;
    }
}

class NewMessageEvent {
    constructor(message, from, sent) {
        this.message = message;
        this.from = from;
        this.sent = sent;
    }
}

class ChangeChatRoomEvent {
    constructor(name) {
        this.name = name;
    }
}

class ChangeUserChatEvent {
    constructor(name) {
        this.name = name;
    }
}

File: examples/in_memory/events.go
package main

import "time"

const (
	// EventBroadcastMessage is the event name for new chat messages to be broadcasted to all other clients
	EventBroadcastMessage = "broadcast_message"
	// EventSendMessage is the event name for new chat messages sent
	EventSendMessage = "send_message"
	// EventNewMessage is a response to send_message
	EventNewMessage = "new_message"
	// EventChangeSession is event when switching rooms
	EventChangeSession = "change_session"
)

type BroadcastMessageEvent struct {
	Message string `json:"message"`
	From    string `json:"from"`
}

type NewMessageEvent struct {
	Message string    `json:"message"`
	From    string    `json:"from"`
	Sent    time.Time `json:"sent"`
}

type SendMessageEvent struct {
	Message string `json:"message"`
	From    string `json:"from"`
	To      string `json:"to"`
}

type ChangeSessionEvent struct {
	Name string `json:"name"`
}
